blocks/library/text/index.js

```diff
@@ -48,40 +45,46 @@ registerBlock( 'core/text', {
 	],
 
 	edit( { attributes, setAttributes, insertBlockAfter } ) {
-		const { content, align } = attributes;
+		const { content = <p />, align } = attributes;
 
 		return (
 			<Editable
-				value={ fromValueToParagraphs( content ) }
-				onChange={ ( paragraphs ) => setAttributes( {
-					content: fromParagraphsToValue( paragraphs )
-				} ) }
+				value={ content }
+				onChange={ ( nextContent ) => {
+					setAttributes( {
+						content: nextContent
+					} );
+				} }
 				style={ align ? { textAlign: align } : null }
 				onSplit={ ( before, after ) => {
-					setAttributes( { content: fromParagraphsToValue( before ) } );
+					setAttributes( { content: before } );
 					insertBlockAfter( wp.blocks.createBlock( 'core/text', {
-						content: fromParagraphsToValue( after )
+						content: after
 					} ) );
 				} }
 			/>
 		);
 	},
 
 	save( { attributes } ) {
-		const { align, content } = attributes;
+		let { content } = attributes;
+		if ( ! Array.isArray( content ) ) {
```

Comment:
>Could this be something different than an array of undefined ?

Yes, I've added some clarifying comments and slightly adjusted the behavior to return early on an `undefined` content in the rebased d4f5a57c4854c11d405691b37824e63cefb3eab1.

Specifically:

```js
// Content can be in the following shapes, so we normalize to an array:
// - Single paragraph: Object
// - Multiple paragraph: Array of objects
```